src/coreComponents/common/format/table/TableData.cpp

```diff
@@ -59,13 +101,31 @@ void TableData2D::collectTableValues( arraySlice1d< real64 const > dim0AxisCoord
   arraySlice1d< real64 const > columAxisCoordinates = columnMajorInputValues ? dim0AxisCoordinates : dim1AxisCoordinates;
   integer const nCol = columAxisCoordinates.size();
   integer const nRow = rowAxisCoordinates.size();
-  // TODO: 1. restore the table non-blocking error system. 2. add this assert 3. add any other error to it.
-  GEOS_ASSERT( nRow * nCol == values.size() );
+
+  array1d< real64 > wellConstructValues( values.size() );
+
+  if( nRow * nCol != values.size())
+  {
+    wellConstructValues = values;
+    if( nRow * nCol > values.size())
+    {
+      m_errors->addError( GEOS_FMT( "Warning : The number of values is insufficient for the number of columns * rows:\n"
+                                    "Expected {} values, Found {} values", nRow * nCol, values.size() ) );
+      wellConstructValues.resizeDefault( nRow * nCol, 0 );
+    }
+    else
+    {
+      m_errors->addError( GEOS_FMT( "Warning : Too much data for the number of columns * rows:\n"
+                                    "Expected {} values, Found {} values\n"
+                                    "Data may be missaligned", nRow * nCol, values.size() ) );
+    }
```

Comment:
Suggestion of message here:
```suggestion
    if( values.size() < nRow * nCol )
    {
      m_errors->addError( GEOS_FMT( "Warning: Not enough for the number of columns & rows:\n"
                                    "  - Expected {} values ({} columns x {} rows),\n  - Found {} values",
                                    nRow * nCol, nCol, nRow, values.size() ) );
      wellConstructValues.resizeDefault( nRow * nCol, 0 );
    }
    else
    {
      m_errors->addError( GEOS_FMT( "Warning: Too much data for the number of columns & rows:\n"
                                    "  - Expected {} values ({} columns x {} rows),\n  - Found {} values",
                                    "Data may be missaligned",
                                    nRow * nCol, nCol, nRow, values.size() ) );
    }
```